src/ir/analysis/derive.rs

```diff
@@ -143,7 +143,10 @@ impl<'ctx> CannotDerive<'ctx> {
                 "    cannot derive {} for blacklisted type",
                 self.derive_trait
             );
-            return CanDerive::No;
+            match self.derive_trait {
```

Comment:
This seems like a bit of a hack, but it does represent the actual behavior. Before this patch, a CanDerive::No for Default implies that needs_default_impl is still true in codegen/mod.rs as long as the derive_default option is enabled and the type is not a forward declaration. So changing it Manually for blacklisted types doesn't change the behavior as far as I can tell.

Reply:
I don't think the previous behavior made particular sense... I'd be fine with making the behavior change here so all derive traits are consistent.